### PR TITLE
Improved validation of component_id field

### PR DESCRIPTION
**Pull request by jbrooksuk**

### CODE REVIEW

File: app/Commands/Incident/ReportIncidentCommand.php [hunk 1]
```diff
@@ -93,7 +93,7 @@
         'status'           => 'required|int|min:0|max:4',
         'message'          => 'string',
         'visible'          => 'bool',
-        'component_id'     => 'int',
+        'component_id'     => 'int|exists:components|required_with:component_status',
```
Comment (GrahamCampbell) [inline app/Commands/Incident/ReportIncidentCommand.php]: > exists:components

What does that do?

Comment (jbrooksuk) [inline app/Commands/Incident/ReportIncidentCommand.php]: Checks that the component actually exists.

**Comment by GrahamCampbell** [comment]
Please rebase this. ;P

**Comment by jbrooksuk** [comment]
Rebased.

File: app/Commands/Incident/ReportIncidentCommand.php (hunk 1 above)
Comment (GrahamCampbell) [inline app/Commands/Incident/ReportIncidentCommand.php]: I thought we agreed not to make database queries in the validator?

Comment (jbrooksuk) [inline app/Commands/Incident/ReportIncidentCommand.php]: Good point, we did.

Comment (GrahamCampbell) [inline app/Commands/Incident/ReportIncidentCommand.php]: Just done a quick search and `exists:` doesn't match anything in our codebase.

**Comment by GrahamCampbell** [comment]
I've added issue closing data to the description. Is that correct?

**Comment by GrahamCampbell** [comment]
Rather than passing through the ID, shouldn't we pass through an actual instance of the model?

**Comment by GrahamCampbell** [comment]
That way we already know it exists, right?

**Comment by jbrooksuk** [comment]
> I've added issue closing data to the description. Is that correct?

No, I don't know WTF is going on in that issue. This is just extra validation.

> Rather than passing through the ID, shouldn't we pass through an actual instance of the model?

Yeah, that's the better way of doing it.

**Comment by GrahamCampbell** [comment]
Ok, I've removed the closes docs.

**Pull request merged**